apps/src/templates/teacherDashboard/SectionRow.jsx

```diff
@@ -208,18 +227,17 @@ class SectionRow extends Component {
 
   render() {
     const {
-      section,
+      sections,
+      sectionId,
       validLoginTypes,
       validGrades,
       validAssignments
     } = this.props;
     const { editing, deleting } = this.state;
 
-    // When deleting a section, I've occasionally seen us attempt a render with
-    // a null section for some reason. This is here to provide defense against this.
-    if (!section) {
-      return null;
-    }
+    const section = sections[sectionId];
```

Comment:
Hmmm, you're right. I am probably able to get into the same situation. I haven't hit it so far, but I also wasnt hitting it consistently before. Might be safest to leave my null check in.